### PR TITLE
Disable ThemeCheck on vite-tag liquid snippets

### PR DESCRIPTION
**Pull request by builtbylane**
We run all liquid through the [Shopify Theme Check](https://shopify.dev/docs/themes/tools/theme-check) linter as part of deploy CI/CD

This prevents the generated snippets from raising any lint errors/warnings.

_Before:_ 
CleanShot 2023-06-16 at 17 49 39

_After:_ 
CleanShot 2023-06-16 at 17 47 12

Thanks for the great project

### CODE REVIEW

File: packages/vite-plugin-shopify/src/html.ts [hunk 1]
```diff
@@ -145,6 +145,12 @@ export default function shopifyHTML (options: Required<Options>): Plugin {
 
 const viteTagDisclaimer = '{% comment %}\n  IMPORTANT: This snippet is automatically generated by vite-plugin-shopify.\n  Do not attempt to modify this file directly, as any changes will be overwritten by the next build.\n{% endcomment %}\n'
 
+// Disable Shopify theme-check linter at top of file
+const themeCheckDisable = '{% comment %}theme-check-disable{% endcomment %}\n'
```
Comment (montalvomiguelo) [inline packages/vite-plugin-shopify/src/html.ts:149]: There are 2 offenses during dev; could we specify those? `AssetUrlFilters,RemoteAsset`

File: packages/vite-plugin-shopify/src/html.ts [hunk 2]
```diff
@@ -135,7 +135,7 @@ export default function shopifyHTML (options: Required<Options>): Plugin {
         }
       })
 
-      const viteTagSnippetContent = viteTagDisclaimer + viteTagEntryPath(config.resolve.alias, options.entrypointsDir, viteTagSnippetName) + assetTags.join('\n') + '\n{% endif %}\n'
+      const viteTagSnippetContent = viteTagDisclaimer + themeCheckDisable + viteTagEntryPath(config.resolve.alias, options.entrypointsDir, viteTagSnippetName) + assetTags.join('\n') + '\n{% endif %}\n' + themeCheckEnable
```
Comment (montalvomiguelo) [inline packages/vite-plugin-shopify/src/html.ts:138]: There are 0 offenses during prod; no need to disable checks here

**Review by montalvomiguelo — CHANGES_REQUESTED** [review]

File: packages/vite-plugin-shopify/src/html.ts (hunk 1 above)
Comment (builtbylane) [reply packages/vite-plugin-shopify/src/html.ts]: updated with your feedback. thanks

Comment (montalvomiguelo) [reply packages/vite-plugin-shopify/src/html.ts]: I would love your thoughts on not turning off these checks so we can catch the issue and then build production assets for deployment.

**Comment by montalvomiguelo** [comment]
I would love your thoughts on not turning off these checks so we can catch the issue and then build production assets for deployment (:

**Comment by wturnerharris** [comment]
@builtbylane - thanks for your contribution. We discussed and think the production build should be the final asset that you run the Shopify Theme Check against since when running in the dev/local context, the tags are specific to the development process and are not meant to be run against any validations. We will close this PR at this time. Thanks again!

**Pull request closed without merging**